community_tasks/ger_rag_eval.py

```diff
@@ -0,0 +1,224 @@
+from lighteval.metrics.utils import MetricCategory, MetricUseCase
+from lighteval.tasks.lighteval_task import LightevalTaskConfig
+from lighteval.tasks.requests import Doc
+from lighteval.tasks.tasks_prompt_formatting import LETTER_INDICES
+
+
+task1 = LightevalTaskConfig(
+    name="ger_rag_eval_task1",
+    prompt_function="prompt_fn_task1",
+    suite=["community"],
+    hf_repo="deutsche-telekom/Ger-RAG-eval",
+    hf_subset="task1",
+    hf_avail_splits=["test"],
+    evaluation_splits=["test"],
+    few_shots_split="test",
+    few_shots_select="sequential",
+    metric=["loglikelihood_acc"],
+)
+
+task2 = LightevalTaskConfig(
+    name="ger_rag_eval_task2",
+    prompt_function="prompt_fn_task2",
+    suite=["community"],
+    hf_repo="deutsche-telekom/Ger-RAG-eval",
+    hf_subset="task2",
+    hf_avail_splits=["test"],
+    evaluation_splits=["test"],
+    few_shots_split="test",
+    few_shots_select="sequential",
+    metric=["loglikelihood_acc"],
+)
+
+task3 = LightevalTaskConfig(
+    name="ger_rag_eval_task3",
+    prompt_function="prompt_fn_task3",
+    suite=["community"],
+    hf_repo="deutsche-telekom/Ger-RAG-eval",
+    hf_subset="task3",
+    hf_avail_splits=["test"],
+    evaluation_splits=["test"],
+    few_shots_split="test",
+    few_shots_select="sequential",
+    metric=["loglikelihood_acc"],
+)
+
+task4 = LightevalTaskConfig(
+    name="ger_rag_eval_task4",
+    prompt_function="prompt_fn_task4",
+    suite=["community"],
+    hf_repo="deutsche-telekom/Ger-RAG-eval",
+    hf_subset="task4",
+    hf_avail_splits=["test"],
+    evaluation_splits=["test"],
+    few_shots_split="test",
+    few_shots_select="sequential",
+    metric=["loglikelihood_acc"],
+)
+
+QUERY_TASK1: str = """\
+Welche der folgenden Fragen (A oder B oder C oder D) lässt sich anhand des Kontext beantworten?
+
+Kontext:
+{context}
+
+Fragen:
+A: {choice_a}
+B: {choice_b}
+C: {choice_c}
+D: {choice_d}
+"""
+
+
+QUERY_TASK2: str = """\
+Auf Basis welcher der folgenden Kontexte (A oder B oder C oder D) lässt sich die Frage beantworten?
+
+Frage: {question}
+
+Kontexte:
+
+A:
+{choice_a}
+
+B:
+{choice_b}
+
+C:
+{choice_c}
+
+D:
+{choice_d}
+"""
+
+
+QUERY_TASK3: str = """\
+Beantwortet die Antwort wirklich die Frage?
+Antworte mit J für ja oder N für nein.
+
+Die Frage: {question}
+
+Die Antwort: {answer}
+"""
+
+
+QUERY_TASK4: str = """\
+Lässt sich die Frage mithilfe der Informationen aus dem Kontext beantworten?
+Antworte mit J für ja oder N für nein.
+
+Kontext:
+{context}
+
+Die Frage: {question}
+"""
+
+
+def prompt_fn_task1(line, task_name: str = None):
+    query = QUERY_TASK1.format(
+        context=line["context"],
+        choice_a=line["choice_a"],
+        choice_b=line["choice_b"],
+        choice_c=line["choice_c"],
+        choice_d=line["choice_d"],
+    )
+    choices = ["A", "B", "C", "D"]
+    return Doc(
+        task_name=task_name,
+        query=query,
+        choices=choices,
+        gold_index=choices.index(line["target"]),
```

Comment:
You need to add `Auf Basis welcher der folgenden Kontexte (A oder B oder C oder D) lässt sich die Frage beantworten?\n\n` to the instruction parameter of the Doc item, and same for all the other instructions. That way, the line does not get duplicated n times if the task is launched in n-shot

Reply:
fixed